### ISSUE TITLE
[webhook] Request limits

### ISSUE THREAD

**Issue by ParkMyCar** [MaterializeInc/materialize#20227]
### Feature request

As a form of protection, we need to impose limits around the new `/api/webhook` endpoint. Two things called out in the design doc were:

1. Number of requests per-second. We'll set a default limit to 100, but it should be configurable via a system parameter.
2. Size of request. We'll set a default limit of 16KiB, but it should be configurable via a system parameter.

In the future ideally these limits exist per-webhook, but that's future work and out of scope for this task.

**Comment by benesch** [comment]
Sorry to be late to the party here, but wanted to point you at a similar PR for MaterializeInc/cloud: https://github.com/MaterializeInc/cloud/pull/6497#pullrequestreview-1447923776

We ended up using [`tower_governor`](https://crates.io/crates/tower_governor) there. (As you discovered, tower's built-in rate limit and load shed layers are kinda wonky to stick together). No hard reason we need to use the same crate across the codebases, but @ParkMyCar I'd love to have you look at `tower_governor` and see if it makes sense to use in `environmentd`, too. It would let us do separate limits per-IP, which might be useful in the future. Though I'm not sure whether it does the thing you wanted of limiting outstanding requests, vs just limiting the rate of _incoming_ requests.

**Comment by ParkMyCar** [comment]
Thanks for pointing out `tower_governor`! I came across it when I was looking for alternatives to the traditional `tower` layers, but it couldn't limit concurrent requests, instead it rate limits based on time, which is why I didn't end up using it.

I filed https://github.com/MaterializeInc/materialize/issues/20876 for use in `environmentd` though! I think having a global rate limit for all the end points (albeit a high one) makes a lot of sense.

As I called out in that comment, `tower_governor` seems to work a bit different than other `tower` layers, which shouldn't effect correctness or performance at all, but might make it incompatible with other `tower` layers.
